File: strategy/marvin/agent/defender/penalty.lua
local Base = require "agent/base/behavior"
local Penalty = Class("Agent.Defender.Penalty", Base)

local World = require "../base/world"
local DefendPenalty = require "task/defender/defendpenalty"


function Penalty:check()
	return World.RefereeState == "PenaltyDefensivePrepare" or World.RefereeState == "PenaltyDefensive"
end

function Penalty:_updateTask()
	return DefendPenalty
end

return Penalty

File: strategy/typescript/glados/agent/defender/penalty.ts
import * as World from "base/world";
import {Behavior} from "glados/agent/base/behavior";
import {Task} from "glados/task/base";
import {DefendPenalty} from "glados/task/defender/defendpenalty";

export class Penalty extends Behavior {
	check (): boolean {
		return World.RefereeState === "PenaltyDefensivePrepare" || World.RefereeState === "PenaltyDefensive";
	}

	_updateTask (): [typeof Task] {
		return [DefendPenalty];
	}
}